### PR TITLE
Onix transform fixes

### PR DESCRIPTION
**Pull request by ssairanen**
Transform mostly fixed
(Comma missing after a$ subfield in 100/700 fields, some other ending punctuation validation problems)

### CODE REVIEW

File: src/transform.js [hunk 1]
```diff
@@ -338,28 +425,34 @@ export default async stream => {
...
-
-					return field;
-				}
-
-				field.subfields = [{code: 'a', value: start}];
-				return field;
+			const results = [/\s+[\u2013\u2014-]\s+/.exec(title),/:\s+|\.+/.exec(title),/!+|\?+/.exec(title)];
+			const slices = {start: "", end: ""};
+
+			if (results[0]) {
```
Comment (natlibfi-arlehiko) [inline src/transform.js]: What the if and if-elses in a function and do it like this:
```js
if (result[0]) {
  return foo();
}

field.subfields = [{code: 'a', value: title}];
return field;
```

File: src/transform.js [hunk 2]
```diff
@@ -338,28 +425,34 @@ export default async stream => {
...
-						{code: 'a', value: `${start} :`},
-						{code: 'b', value: end}
-					];
-
-					return field;
-				}
-
-				field.subfields = [{code: 'a', value: start}];
-				return field;
+			const results = [/\s+[\u2013\u2014-]\s+/.exec(title),/:\s+|\.+/.exec(title),/!+|\?+/.exec(title)];
```
Comment (natlibfi-arlehiko) [inline src/transform.js]: Add a comment which tells what characters those codes represent.

File: src/transform.js [hunk 3]
```diff
@@ -581,4 +666,25 @@ export default async stream => {
 			}, str.split('')).join('');
 		}
 	}
+
+	function dropRecord(node) {
+		return node.children['ProductIdentifier'].reduce((acc, n) => {
```
Comment (natlibfi-arlehiko) [inline src/transform.js]: Use [Array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).

File: src/transform.js [hunk 4]
```diff
@@ -26,6 +26,7 @@
 *
 */
 
+import http from 'http';
```
Comment (natlibfi-arlehiko) [inline src/transform.js]: We use [node-fetch](https://www.npmjs.com/package/node-fetch).

File: src/transform.js [hunk 5]
```diff
@@ -238,14 +244,96 @@ export default async stream => {
...
+				{code: 'a', value: 'Aineisto on käytettävissä tutkimus- ja muihin tarkoituksiin;'},
+				{code: 'b', value: 'Kansalliskirjasto;'},
+				{code: 'c', value: 'Laki kulttuuriaineistojen tallettamisesta ja säilyttämisestä'},
+				{code: 'u', value: 'http://www.finlex.fi/fi/laki/ajantasa/2007/20071433 '},
+				{code: '5', value: 'FI-Vapaa'},
+				{code: '9', value: 'FENNI<KEEP>'}
+			]
+		});
+
+		(async() => {
```
Comment (natlibfi-arlehiko) [inline src/transform.js]: Do it like this:
```js
await createURN();
```

File: src/transform.js [hunk 6]
```diff
@@ -238,14 +244,96 @@ export default async stream => {
...
+				{code: 'k', value: isil},
+				{code: 'q', value: 'FI-NL'}
+				
+			]
+		});
+
 		return record;
 
+		function createURN() {
+			if (isbn) {
```
Comment (natlibfi-arlehiko) [inline src/transform.js]: Do the implementation like this:
```js
if (isbn) {
  return `http://urn.fi/URN:ISBN:${isbn}`
}

const response = await fetch(URL_GENERATOR_URL);
```

Declare URL_GENERATOR_URL at the top of the file where other consts are declacred as well.

File: src/transform.js (hunk 1 above)
Comment (ssairanen) [reply src/transform.js]: done?

File: src/transform.js (hunk 3 above)
Comment (ssairanen) [reply src/transform.js]: done

File: src/transform.js (hunk 2 above)
Comment (ssairanen) [reply src/transform.js]: done

File: src/transform.js (hunk 4 above)
Comment (ssairanen) [reply src/transform.js]: done

File: src/transform.js (hunk 6 above)
Comment (ssairanen) [reply src/transform.js]: Is it possible to make fetch/async code work inside of a stream? The field doesn't seem to be included in the record if fetch is used.

Comment (ssairanen) [reply src/transform.js]: moved to validator

**Pull request merged**